UnknownMetadataKey for created_at

Hey, I’m using rvx v0.5.3 to run the puppet-lint gem in my puppet repo. I’m encountering a repeated error that is ~new in the last week:

```
Run rvx puppet-lint \
  rvx puppet-lint \
    --fail-on-warnings \
    --no-parameter_order-check \
    --code_on_top_scope-check \
    code/modules
  shell: /usr/bin/bash -e {0}
Error: ToolError(ToolRunError(Install(GemReleaseParse(UnknownMetadataKey { key: "created_at", metadata: "created_at:2011-08-15T18:41:56Z" }))))
Error: Process completed with exit code 1.
```

This was running the puppet-lint binary fine before. `rv ci` is fine since the gems are cached.

I expected rvx to execute puppet-lint without error.

1. Would it be possible to make the error message clearer on next steps? Should I try removing gems? Clearing cache? Which gem is causing the issue in the first place?
2.  Would you consider supporting this metadata, `created_at`?

Thanks!

--- a/crates/rv/src/gemserver.rs
+++ b/crates/rv/src/gemserver.rs
@@ -519,6 +519,9 @@ mod tests {
     fn test_unknown_and_created_at_metadata() {
         let input = "1.0.0 |checksum:505c6770a5ec896244d31d7eac08663696d22140493ddb820f66d12670b669d2,created_at:2011-08-15T18:41:56Z,unknown_key_to_ignore:foo";
         let actual = GemRelease::parse(input).unwrap();
-        assert_eq!(actual.metadata.created_at, Some("2011-08-15T18:41:56Z".to_string()));
+        assert_eq!(
+          actual.metadata.created_at,
+          Some("2011-08-15T18:41:56Z".to_string())
+        );
     }
 }

--- a/crates/rv/src/gemserver.rs
+++ b/crates/rv/src/gemserver.rs
@@ -520,8 +520,8 @@ mod tests {
         let input = "1.0.0 |checksum:505c6770a5ec896244d31d7eac08663696d22140493ddb820f66d12670b669d2,created_at:2011-08-15T18:41:56Z,unknown_key_to_ignore:foo";
         let actual = GemRelease::parse(input).unwrap();
         assert_eq!(
-          actual.metadata.created_at,
-          Some("2011-08-15T18:41:56Z".to_string())
+            actual.metadata.created_at,
+            Some("2011-08-15T18:41:56Z".to_string())
         );
     }
 }

--- a/crates/rv/src/gemserver.rs
+++ b/crates/rv/src/gemserver.rs
@@ -369,11 +369,11 @@ fn parse_metadata(metadata: &str) -> ParseResult<Metadata> {
             "published_at" => {
                 //Unused for now
             }
+            "created_at" => {
+                out.created_at = Some(v.to_owned());
+            }
             _ => {
-                return Err(GemReleaseParse::UnknownMetadataKey {
-                    key: k.to_owned(),
-                    metadata: md_str.to_owned(),
-                });
+                // Ignore other fields in the future
             }
         }
     }
@@ -404,6 +404,7 @@ pub struct Metadata {
     pub checksum: Vec<u8>,
     pub ruby: Requirement,
     pub rubygems: Requirement,
+    pub created_at: Option<String>,
 }
 
 impl Default for Metadata {
@@ -416,6 +417,7 @@ impl Default for Metadata {
             rubygems: Requirement {
                 constraints: vec![],
             },
+            created_at: None,
         }
     }
 }
@@ -426,6 +428,7 @@ impl std::fmt::Debug for Metadata {
             .field("checksum", &hex::encode(&self.checksum))
             .field("ruby", &self.ruby)
             .field("rubygems", &self.rubygems)
+            .field("created_at", &self.created_at)
             .finish()
     }
 }
@@ -510,5 +513,12 @@ mod tests {
                 .to_string(),
             expected_release
         );
+    }
+
+    #[test]
+    fn test_unknown_and_created_at_metadata() {
+        let input = "1.0.0 |checksum:505c6770a5ec896244d31d7eac08663696d22140493ddb820f66d12670b669d2,created_at:2011-08-15T18:41:56Z,unknown_key_to_ignore:foo";
+        let actual = GemRelease::parse(input).unwrap();
+        assert_eq!(actual.metadata.created_at, Some("2011-08-15T18:41:56Z".to_string()));
     }
 }

